specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-09-02-preview/fleets.json

```diff
@@ -0,0 +1,852 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FleetNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/FleetCredentialResults"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List Fleet Credentials": {
+            "$ref": "./examples/Fleets_ListCredentialsResult.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Fleet": {
+      "type": "object",
+      "properties": {
+        "etag": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource Etag."
+        },
+        "properties": {
+          "description": "Properties of a Fleet.",
+          "$ref": "#/definitions/FleetProperties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "The Fleet resource which contains multiple Kubernetes clusters as its members."
+    },
+    "FleetProperties": {
+      "type": "object",
+      "description": "Properties of a Fleet.",
+      "properties": {
+        "hubProfile": {
+          "description": "The FleetHubProfile configures the Fleet's hub.",
+          "$ref": "#/definitions/FleetHubProfile"
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "x-ms-enum": {
+            "name": "FleetProvisioningState",
+            "modelAsString": true
+          },
+          "enum": [
+            "Succeeded",
+            "Failed",
+            "Canceled",
+            "Creating",
+            "Deleting",
+            "Updating"
+          ],
+          "description": "The provisioning state of the last accepted operation."
+        }
+      }
+    },
+    "FleetHubProfile": {
+      "type": "object",
+      "description": "The FleetHubProfile configures the fleet hub.",
+      "properties": {
+        "dnsPrefix": {
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "DNS prefix used to create the FQDN for the Fleet hub.",
+          "pattern": "^[a-zA-Z0-9]$|^[a-zA-Z0-9][a-zA-Z0-9-]{0,52}[a-zA-Z0-9]$"
```

Comment:
Is there a maxLength here of 54 characters?

If so, consider specifying that here too?
```
      "minLength": 1,
      "maxLength": 54,
```

One side of the regex implies there's a max length while the other doesn't really.

Reply:
Oh wait, one side of this regex is just `^[a-zA-Z0-9]$`... that's just a single character. Should that be `^[a-zA-Z0-9]+$`?

Reply:
hm, yes, that's odd there should definitely be a max length too. this mirrors AKS dnsprefix and is passed down as is to the creation of the cluster, so it should be the same constraints.
AKS does not have a swagger constraints though...

Reply:
"A valid DNS prefix must contain between 1 and 54 characters and can contain only letters, numbers, and hyphens. It must start and end with alphanumeric values."

Reply:
the first part is for single character, it cannot be a hyphen.
if it's more than 1 char, then the regex apply.
please add the minLengh and maxLength too @minhng22

Reply:
Added.